client/src/components/history/Weekly.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from "react";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import MyBarChart from "./MyBarChart";
+import MyLineChart from "./MyLineChart";
+import "./Weekly.scss";
+import { apiGet } from "../../api";
+import dayjs from "dayjs";
+
+interface IDaliy {
+  user_id: string;
+  waterAmount: number;
+  weightAmount: number;
+  caloriesAmount: number;
+  foodItems?: any;
+  entryDate?: Date;
+  bmi: number;
+  bfp: number;
+  age?: number;
+}
+
+enum EGender {
+  female = 0,
+  male = 1,
+}
+
+interface IUser {
+  gender: EGender;
+  weight: number;
+  height: number;
+  caloriesGoal: number;
+  waterGoal: number;
+  age: number;
+}
+
+const Weekly = () => {
+  const [userInfo] = useState<IUser>({
+    gender: EGender.female,
+    weight: 50,
+    height: 1.7,
+    caloriesGoal: 500,
+    waterGoal: 8,
+    age: 27,
+  });
+  const [weeklyData, setWeeklyData] = useState<IDaliy[]>([]);
+
+  useEffect(() => {
+    getWeekly();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userInfo]);
+
+  const getWeekly = async () => {
+    //发起请求 ， 成功后根据获取的数据计算bmi，bfp
+    const result = await apiGet("/api/getWeekly");
+    console.log(result);
+    if (result) {
+      const tempData = [...result];
+      const {height, gender, age } = userInfo;
+      tempData.forEach((item) => {
+        item.entryDate = dayjs(item.entryDate).format("MM/DD");
+        item.bmi = Number((item.weightAmount / (height * height)).toFixed(2));
+        if (gender === EGender.female) {
+          item.bfp = 1.2 * item.bmi + 0.23 * age - 5.4;
+        } else if (gender === EGender.male) {
+          item.bfp = 1.2 * item.bmi + 0.23 * age - 16.2;
+        }
+      });
+      setWeeklyData(tempData);
+    }
+  };
+
+  return (
+    <Box className="weekly">
+      <Box className="calories chart_item">
+        <Typography className="title">Calories</Typography>
+        <Box className="chart_wrap">
+          <MyBarChart
+            data={weeklyData.map((item) => ({ value: item.caloriesAmount }))}
```

Comment:
@nicoleqi7 , I see you're simply putting a calorieAmount in the dailyEntries, for now that's fine. You will need to modify your getweekly to crunch the data from the Food items once they are available.